### PR TITLE
Enable widget customization

### PR DESCRIPTION
**Pull request by krschau**
## Summary of the pull request
Implement customization enabled by https://github.com/microsoft/devhome/pull/1819

When the user selects "Customize" in the widget menu, the widget will switch to its customization flow. From there, they can edit the data they put in during the "add" flow and save, or cancel and revert everything back to the way the widget was before.

By implementing the IWidgetProvider2 interface provided by WinAppSDK 1.4, the widget can opt into receiving notifications when the user selects the "Customize" menu, and provide this experience. Otherwise, customization menu item is not available.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: src/GitHubExtension/Widgets/GitHubIssuesWidget.cs [hunk 1]
```diff
@@ -206,9 +213,22 @@ private void DataManagerUpdateHandler(object? source, DataManagerUpdateEventArgs
             if (fullName == e.Description && e.Context.Contains("Issues"))
             {
                 Log.Logger()?.ReportInfo(Name, ShortId, $"Received matching repository update event.");
+
+                // Don't update if we're in configuration mode.
```
Comment (EricJohnson327) [inline src/GitHubExtension/Widgets/GitHubIssuesWidget.cs:217]: Can this just move up so we aren't doing the extra string checks?

Comment (EricJohnson327) [reply src/GitHubExtension/Widgets/GitHubIssuesWidget.cs:217]: Same with others

**Review by EricJohnson327 — APPROVED** [review]

**Review by vineeththomasalex — APPROVED** [review]

File: src/GitHubExtension/Widgets/GitHubIssuesWidget.cs (hunk 1 above)
Comment (krschau) [reply src/GitHubExtension/Widgets/GitHubIssuesWidget.cs]: Done

**Comment by krschau** [comment]
Duplicate of #113

**Pull request merged**